pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/jstl/CustomTypeConverter.java

```diff
@@ -28,31 +30,31 @@
 import java.time.temporal.TemporalAccessor;
 import java.util.Date;
 import java.util.Map;
-import javax.el.ELException;
 import org.apache.avro.util.Utf8;
+import org.apache.el.lang.ELSupport;
+import org.apache.el.util.MessageFactory;
 import org.apache.pulsar.client.api.Schema;
 
 /**
  * Overrides the default TypeConverter coerce to support null values & non-EL coercions (e.g.
  * date/time types) schemas.
  */
-public class CustomTypeConverter extends TypeConverterImpl {
+public class CustomTypeConverter extends TypeConverter {
 
   public static final CustomTypeConverter INSTANCE = new CustomTypeConverter();
 
   private CustomTypeConverter() {
     super();
   }
 
-  @Override
+  @SuppressFBWarnings("NP_BOOLEAN_RETURN_NULL")
   protected Boolean coerceToBoolean(Object value) {
     if (value instanceof byte[]) {
       return Schema.BOOL.decode((byte[]) value);
     }
-    return super.coerceToBoolean(value);
+    return null;
```

Comment:
I'd rather call `ELSupport.coerceToBoolean` than return null

Reply:
what type are we expecting here ?
only byte[] ?

Reply:
> I'd rather call ELSupport.coerceToBoolean than return null

the issue with that is `tomcat-embeded-el` expects `null` and `context.setPropertyResolved(false);` in order to proceed with other resolvers. So I was thinking to set all `coerceTo*` methods to private, and have a single entry point to this class which is the `convert` methods. It will handle two cases:
1) When used with `compute`, it sets `context.setPropertyResolved(false);` and propagates the null
2) When without context, like in `cast`, it returns `ELSupport.coerceToType`

Reply:
> what type are we expecting here ?
only byte[] ?

anything that reasonably maps to boolean (mainly string `"true|false"` or literal `true|false`. This will be taken care of the built in converter (in JUEL or tomcat embedded). Any custom case (like byte[]) we handle our selves like in this example, via the `PULSAR.BOOL` schema decoder...

Reply:
I think it's OK to call ELSupport.coerceToBoolean. It will resolve the property which is exactly what we want. It doesn't matter if it's done now or later.

Reply:
If we do that, we'll basically overwrite the underling order by which the resolvers are invoked. Here is how resolvers are registered: https://github.com/apache/tomcat/blob/c2ad60927672c0c2080bdfbf8274b152fa630312/java/jakarta/el/StandardELContext.java#L53-L58

Also the way to register custom types is via registering ELResolver to the ELContext (notice that [TypeConverter](https://github.com/apache/tomcat/blob/d27c63055a679fb461f88ed07acd9b6962fac187/java/jakarta/el/TypeConverter.java#L22) extends ELResolver resolver and type converter implement . If I simply return EL support (from the code path of compute step), this code will end up in infinite loop:
https://github.com/apache/tomcat/blob/d27c63055a679fb461f88ed07acd9b6962fac187/java/jakarta/el/CompositeELResolver.java#L150

Reply:
if return null is confusing, I can separate the code paths for EL context aware (like compute step) and context non-aware (for cast, or direct invocation of the converter from utility methods) paths.